From 6fb38193ea80354730e5468311412d7e2cbdb9cb Mon Sep 17 00:00:00 2001
From: Vinayakumar B <vinayakumarb@apache.org>
Date: Wed, 5 Jul 2017 16:06:19 +0530
Subject: [PATCH 2552/2748] HADOOP-13414. Hide Jetty Server version header in
 HTTP responses. Contributed by Surendra Singth
 Lilhore.

(cherry picked from commit 1e8301b215f79d48bbb58e2348945144d4c37997)
(cherry picked from commit cd6ee04274cfd14813bf0c492d107c5a3008336a)

Change-Id: I5ea6266b393fe35f654e17c03fc4606f805203fd
---
 .../java/org/apache/hadoop/http/HttpServer2.java   |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
index 6426894..7a009af 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java
@@ -399,6 +399,7 @@ private void initializeWebServer(String name, String hostName,
         : new QueuedThreadPool(maxThreads);
     threadPool.setDaemon(true);
     webServer.setThreadPool(threadPool);
+    webServer.setSendServerVersion(false);
 
     SessionManager sm = webAppContext.getSessionHandler().getSessionManager();
     if (sm instanceof AbstractSessionManager) {
-- 
1.7.9.5

